Document/0x06d-Testing-Data-Storage.md

```diff
@@ -761,16 +761,53 @@ This will cause the background image to be set to the "overlayImage.png" instead
+
+_Swift_ collections (`Array`, `Set` and `Dictionary`), on another hand, might be acceptable as long as they collect primitive data types such as `char` or `int` and are defined as mutable (variables instead of constants), in which case they are more or less equivalent to a primitive array (like `char []`). Still, these collections provide safe memory management, which can result in unidentified copies of the sensitive data in memory, in case the collection needs to copy the underlying buffer to a different location in order to extend it.
+
+Usage of _Objective-C_ mutable data types, such as `NSMutableString` might also be acceptable, but suffer from the same "safe memory" issue as _Swift_ collections. Additionally, attention should be payed when using _Objective-C_ collections as they hold data by reference and only _Objective-C_ data types are allowed. Therefore, we are not looking for a mutable collection, but a collection that references mutable objects.
+
+As we've seen so far usage of _Swift_ or _Objective-C_ data types require deep understanding of the language implementation itself. Further, we have witnessed some core re-factoring between mayor _Swift_ versions, resulting in incompatible behavior of many data types. In order to avoid these issues we recommend usage of primitive data types whenever certain data needs to be securely erased from memory.
+
+Unfortunately, not many libraries and frameworks are designed to allow overwriting of sensitive data. Even Apple does not consider this issue in the official API of the iOS SDK. For example, most of the APIs for data transformation (passers, serializes, etc.) operate on non-primitive data types. Similarly, regardless of weather you flag certain `UITextField` as _Secure Text Entry_ or not, it always return data as `String` or `NSString`.
 
+To summarize, when performing static analysis for sensitive data exposed in memory, you should:
+- Try to identify application components and make a map of where certain data is used.
+- Verify that sensitive data is handled in as few components as possible.
+- Verify that object references are properly removed, once the object containing sensitive data is no longer needed.
+- For highly sensitive information, verify data is overwritten as soon as it is no longer needed.
+  - Such data must not be passed over immutable data types such as `String` or `NSString`.
+  - Non-primitive data types are in general indicative and should be avoided.
+  - Overwriting should be done before removing references.
+  - Pay attention to third-party components (libraries and frameworks).
+    Good indicator is their public API. Is the sensitive data passing the public API handled as proposed in this chapter ? 
```

Comment:
> Good indicator is their public API.

An indicator of what?

> Is the sensitive data passing the public API handled as proposed in this chapter ? 

It took me a couple of reads before I understood this sentence. Maybe you can simplify it a bit?